src/m3ninx/search/searcher/negation.go

```diff
@@ -50,9 +52,24 @@ func (s *negationSearcher) Search(r index.Reader) (postings.List, error) {
 		return nil, err
 	}
 
-	// Perform a lazy fast intersect and negate.
-	// TODO: Try and see if returns err, if so fallback to slower method?
-	intersects := []postings.List{pl}
-	negations := []postings.List{negatePl}
-	return roaring.IntersectAndNegateReadOnly(intersects, negations)
+	if index.MigrationReadOnlyPostings() {
+		// Perform a lazy fast intersect and negate.
+		intersects := []postings.List{pl}
+		negations := []postings.List{negatePl}
+		return roaring.IntersectAndNegateReadOnly(intersects, negations)
+	}
+
+	// Not running migration path, fallback.
+	mutable, ok := pl.(postings.MutableList)
+	if !ok {
+		// Note not creating a "errNotMutable" like error since this path
+		// will be deprecated and we might forget to cleanup the err var.
+		return nil, fmt.Errorf("postings list for non-migration path not mutable")
+	}
+
+	result := mutable.Clone()
+	if err := result.Difference(negatePl); err != nil {
```

Comment:
nit; dont think this cloned before, may be unnecessary if we're falling back anyway?